neurons/validator.py

```diff
@@ -1626,6 +1656,105 @@ async def run(self):
             # 18. Increment global step
             self.global_step += 1
 
+    def select_initial_peers(self) -> tplr.comms.PeerArray | None:
+        try:
+            tplr.logger.info("Starting selection of initial gather peers")
+            # 1. Add top incentive peers
+            uid_to_non_zero_incentive = {
+                uid: incentive
+                for uid, incentive in zip(self.metagraph.uids, self.metagraph.I)
+                if incentive > 0 and uid in self.comms.active_peers
+            }
+            top_incentive_peers = sorted(
+                uid_to_non_zero_incentive,
+                key=uid_to_non_zero_incentive.get,
+                reverse=True,
+            )[: self.hparams.max_topk_peers]
+
+            # Convert to list to ensure it's not a dict_keys object
+            top_incentive_peers = np.array(top_incentive_peers, dtype=np.int64)
```

Comment:
```suggestion
            uid_to_non_zero_incentive = {
                uid: incentive
                for uid, incentive in zip(self.metagraph.uids, self.metagraph.I)
                if incentive > 0 and uid in self.comms.active_peers
            }
            top_incentive_peers = sorted(
                uid_to_non_zero_incentive.items(),
                key=lambda x: x[1],
                reverse=True,
            )[: self.hparams.max_topk_peers]

            # Extract just the UIDs (the first element of each tuple)
            top_incentive_peers = [int(uid) for uid, _ in top_incentive_peers]
```

To me this is a bit clearer what is happening (also using list instead of `np.array` if we decide to go back to that). Wdyt?

Reply:
The lambda thing sure, but skeptical to using list instead of array, replied to that [here](https://github.com/tplr-ai/templar/pull/158#discussion_r2021419577).

Reply:
Lambda nitpick fixed in 9445f69.

Reply:
You would still need to filter out the uid from the tuple with this approach. It is more readable imo when you do it like this.

Reply:
Ignore it and revert if you think it is a nitpick.

Reply:
Damn, reverted in b0697dc.